core/src/main/java/net/opentsdb/data/types/numeric/aggregators/ArrayAverageFactory.java

```diff
@@ -31,73 +34,98 @@
 
   public static final String TYPE = "Avg";
   
+  private ArrayObjectPool int_pool;
+  
   @Override
   public String type() {
     return TYPE;
   }
   
   @Override
   public NumericArrayAggregator newAggregator() {
-    return new ArrayAverage(false);
+    return new ArrayAverage(false, this);
   }
   
   @Override
   public NumericArrayAggregator newAggregator(final AggregatorConfig config) {
-    if (config != null && config instanceof NumericAggregatorConfig) {
-      return new ArrayAverage(((NumericAggregatorConfig) config).infectiousNan());
+    if (config != null && config instanceof NumericArrayAggregatorConfig) {
+      return new ArrayAverage((NumericArrayAggregatorConfig) config, this);
     }
-    return new ArrayAverage(false);
+    return new ArrayAverage(false, this);
   }
   
   @Override
   public NumericArrayAggregator newAggregator(final boolean infectious_nan) {
-    return new ArrayAverage(infectious_nan);
+    return new ArrayAverage(infectious_nan, this);
   }
 
   @Override
   public Deferred<Object> initialize(final TSDB tsdb, final String id) {
     this.id = Strings.isNullOrEmpty(id) ? TYPE : id;
     tsdb.getRegistry().registerPlugin(NumericArrayAggregatorFactory.class, 
         "average", this);
+    setPools(tsdb);
+    int_pool = (ArrayObjectPool) tsdb.getRegistry().getObjectPool(IntArrayPool.TYPE);
     return Deferred.fromResult(null);
   }
   
+  ArrayObjectPool intPool() {
+    return int_pool;
+  }
+  
   public static class ArrayAverage extends BaseArrayAggregator {
-
+    
     protected int[] counts;
+    protected PooledObject int_pooled;
     protected double[] results;
     
-    public ArrayAverage(final boolean infectious_nans) {
-      super(infectious_nans);
+    public ArrayAverage(final boolean infectious_nans,
+                        final BaseArrayFactory factory) {
+      super(infectious_nans, factory);
+    }
+    
+    public ArrayAverage(final NumericArrayAggregatorConfig config,
+                        final BaseArrayFactory factory) {
```

Comment:
`ArrayAverageFactory `

Reply:
Hmm I could have but I'd rather not go back and rebase this whole commit.